obelisk/cpp/hardware/robots/unitree/go2_interface.h

```diff
@@ -267,64 +247,161 @@ namespace obelisk {
+                    dds_low_command.motor_cmd().at(i).mode() = 1;  // 1:Enable, 0:Disable
+                    dds_low_command.motor_cmd().at(i).tau() = 0;
+                    dds_low_command.motor_cmd().at(i).q() = 0;
+                    dds_low_command.motor_cmd().at(i).dq() = 0;
+                    dds_low_command.motor_cmd().at(i).kp() = 0;
+                    dds_low_command.motor_cmd().at(i).kd() = kd_damping_[i];
+                }
+                dds_low_command.crc() = Crc32Core((uint32_t *)&dds_low_command, (sizeof(dds_low_command) >> 2) - 1);
+                lowcmd_publisher_->Write(dds_low_command);
+            }
+        }
 
-        //     // IMU
-        //     obelisk_sensor_msgs::msg::ObkImu obk_imu_state;
-        //     obk_imu_state.header.stamp = this->now();
-        //     obk_imu_state.angular_velocity.x = imu_state.gyroscope()[0];
-        //     obk_imu_state.angular_velocity.y = imu_state.gyroscope()[1];
-        //     obk_imu_state.angular_velocity.z = imu_state.gyroscope()[2];
+        void TransitionToUnitreeVel() override{
+            sport_client_.ClassicWalk(false);
+        }
 
-        //     obk_imu_state.orientation.w = imu_state.quaternion()[0];
-        //     obk_imu_state.orientation.x = imu_state.quaternion()[1];
-        //     obk_imu_state.orientation.y = imu_state.quaternion()[2];
-        //     obk_imu_state.orientation.z = imu_state.quaternion()[3];
+        bool EngageUnitreeMotionControl() override{
+            if (high_level_ctrl_engaged_) {
+                return true;
+            }
+            int32_t ret, status;
+            ret = robot_state_client_.ServiceSwitch("mcf", 1, status);  // Turn the Unitree motion control on
+            if (ret == 0) {
+                high_level_ctrl_engaged_ = true;
+                sport_client_.StopMove();
+            } else {
+                RCLCPP_ERROR_STREAM(this->get_logger(), "ENGAGING UNITREE MOTION CONTROL FAILED! ret = " << ret);
+            }
+            return ret == 0;
+        }
 
-        //     obk_imu_state.linear_acceleration.x = imu_state.accelerometer()[0];
-        //     obk_imu_state.linear_acceleration.y= imu_state.accelerometer()[1];
-        //     obk_imu_state.linear_acceleration.z = imu_state.accelerometer()[2];
+        bool ReleaseUnitreeMotionControl() override{
+            if (!high_level_ctrl_engaged_) {
+                return true;
+            }
+            sport_client_.StopMove();
+            int32_t ret, status;
+            ret = robot_state_client_.ServiceSwitch("mcf", 0, status);  // Turn the Unitree motion control off
+            if (ret == 0) {
+                high_level_ctrl_engaged_ = false;
+            } else {
+                RCLCPP_ERROR_STREAM(this->get_logger(), "RELEASING UNITREE MOTION CONTROL FAILED! ret = " << ret);
+            }
+            return ret == 0;
+        }
+
+        void WriteMotorLogHeader() override{
```

Comment:
Can we move this logging function (and others like the joint name file and writing the data) to the parent class? Is any of it the same? Looks like the headers and joint files might be?